test/blackbox-tests/test-cases/codeberg.t

```diff
@@ -0,0 +1,17 @@
+Test the codeberg source type in project files.
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.17)
+  > (name foo)
+  > (generate_opam_files true)
+  > (source (codeberg john/doe))
+  > (package
+  >  (allow_empty)
+  >  (name foo))
+  > EOF
+
+  $ dune build
+  $ cat foo.opam | grep -i codeberg.org
+  homepage: "https://codeberg.org/john/doe"
+  bug-reports: "https://codeberg.org/john/doe/issues"
+  dev-repo: "git+https://codeberg.org/john/doe.git"
```

Comment:
It would be good to have a test that shows that this doesn't work on dune-lang < 3.17.

Reply:
Added in 207cae89ccb769241f8d7f31f913f0398b095193